File: LEETCODE1600/Output/intermediate/LEETCODE_AST/35-search-insert-position搜索插入位置/c_code/9.c
int searchInsert(int* nums, int numsSize, int target){
    int l=0;
    int r=numsSize-1;
    int m = (l+r)/2;
    if(target < nums[l]) return 0;
    if(target > nums[r]) return numsSize;
    while(r-l>2)
    {
        if(nums[m] > target){
            r = m;
            m = (l+r)/2; 
        }
        else if(nums[m] < target){
            l = m;
            m = (l+r)/2;
        }
        else{
            return m;
        }
    }
    for(int i=l;i<=r;i++){
        if(nums[i] >= target){
            return i;
        }        
    }
    return 0;
}

File: LEETCODE1600/Output/LEETCODE_Data/1-two-sum两数之和/pair_24/9.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut res = vec![];
        let mut cache = HashMap::new();

        for (index, item) in nums.iter().enumerate() {
            let sub = target - *item;

            match cache.get(item) {
                Some(pre_index) => {
                    res.push(*pre_index);
                    res.push(index as i32);
                    return res
                },
                None => {
                    cache.insert(sub, index as i32);
                },
            }
        }
        res
    }
}